### PR TITLE
BUG: Don't raise TypeError when converting NA from string to numeric

### LINKED ISSUE #37262: BUG: to_numeric() raises TypeError for StringDtype series
- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest version of pandas.

- [X] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
import pandas as pd
pd.to_numeric(pd.Series(['1', None, '2'], dtype='string'))
```

#### Problem description

Running this code results in `TypeError: Invalid object type at position 1`

The same code works for old-style str dtype. Running

```python
import pandas as pd
pd.to_numeric(pd.Series(['1', None, '2'], dtype='str'))
```

outputs

```
0    1.0
1    NaN
2    2.0
dtype: float64
```

This is a common way to convert strings to floats, and switching from the old-style 'str' to the new-style 'string' dtype will break code using it.

#### Expected Output

```
0    1.0
1    NaN
2    2.0
dtype: float64
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : db08276bc116c438d3fdee492026f8223584c477
python           : 3.8.2.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 18.7.0
Version          : Darwin Kernel Version 18.7.0: Tue Aug 20 16:57:14 PDT 2019; root:xnu-4903.271.2~2/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.1.3
numpy            : 1.18.5
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 19.2.3
setuptools       : 41.2.0
Cython           : 0.29.21
pytest           : 6.1.1
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.15.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : 0.4.1
gcsfs            : None
matplotlib       : 3.2.2
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 0.17.1
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : 1.3.19
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : 0.50.0

</details>

#### Output of ``pd.show_versions()`` (master)

<details>

INSTALLED VERSIONS
------------------
commit           : c15ded3ed653447415d34b72b62127fa612fd741
python           : 3.8.2.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 18.7.0
Version          : Darwin Kernel Version 18.7.0: Tue Aug 20 16:57:14 PDT 2019; root:xnu-4903.271.2~2/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.2.0.dev0+829.gc15ded3ed
numpy            : 1.18.5
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 19.2.3
setuptools       : 41.2.0
Cython           : 0.29.21
pytest           : 6.1.1
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.15.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : 0.4.1
gcsfs            : None
matplotlib       : 3.2.2
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 0.17.1
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : 1.3.19
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : 0.50.0

</details>

### PR DESCRIPTION
**Pull request by dsaxton**
- [x] closes #37262
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Thanks @dsaxton. It's probably fine to fix it like this in the short term, since a proper fix to actually refactor the `maybe_convert_numeric` to produce nullable results is a bigger change?

**Comment by dsaxton** [comment]
> Thanks @dsaxton. It's probably fine to fix it like this in the short term, since a proper fix to actually refactor the `maybe_convert_numeric` to produce nullable results is a bigger change?

That was my thinking, remove the error now then "enhance" to cast to nullable as a follow-up. I can open a separate issue for that if that makes sense?

File: pandas/tests/tools/test_to_numeric.py [hunk 1]
```diff
@@ -707,3 +707,18 @@ def test_precision_float_conversion(strrep):
     result = to_numeric(strrep)
 
     assert result == float(strrep)
+
+
+@pytest.mark.parametrize(
+    "values, expected",
+    [
+        (["1", "2", None], Series([1, 2, np.nan])),
+        (["1", "2", "3"], Series([1, 2, 3])),
```
Comment (jreback) [inline pandas/tests/tools/test_to_numeric.py:716]: can you add an example that has mixed values (e.g. '3' and 2 and 2.0 as well)

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @dsaxton very nice

**Pull request merged**